src/JoypadModule/src/Module.cpp

```diff
@@ -27,42 +27,49 @@ bool JoypadModule::configure(yarp::os::ResourceFinder &rf)
+
     // set the deadzone interval
-    if(!YarpUtilities::getNumberFromSearchable(rf, "deadzone", m_deadzone))
+    if (!YarpUtilities::getNumberFromSearchable(rf, "deadzone", m_deadzone))
     {
         yError() << "[configure] Unable to get a double from a searchable";
         return false;
     }
 
     // set the maximum value measured by the joypad
-    if(!YarpUtilities::getNumberFromSearchable(rf, "fullscale", m_fullscale))
+    if (!YarpUtilities::getNumberFromSearchable(rf, "fullscale", m_fullscale))
     {
         yError() << "[configure] Unable to get a double from a searchable";
         return false;
     }
 
     const std::string connectPortsTag = "connectPortsSeparately";
-    //connectPortsSeparately is present or set to true
+    // connectPortsSeparately is present or set to true
     m_connectPortsSeparately = rf.check(connectPortsTag) && (rf.find(connectPortsTag).isNull() || rf.find(connectPortsTag).asBool());
 
-    std::vector<std::pair<std::string, int&>> joypadInputs = {{"forwardAxis", m_forwardAxis},
-                                                              {"rotationAxis", m_rotationAxis},
-                                                              {"sideAxis", m_sideAxis},
-                                                              {"prepareButton", m_prepareButton},
-                                                              {"startButton", m_startButton},
-                                                              {"pauseButton", m_pauseButton},
-                                                              {"stopButton", m_stopButton},
-                                                              {"connectGoalButton", m_connectGoalButton},
-                                                              {"connectRpcButton", m_connectRpcButton},
-                                                              {"disconnectButton", m_disconnectButton}};
-    for (auto& input : joypadInputs)
+    std::vector<std::pair<std::string, int &>> joypadInputs = {{"forwardAxis", m_forwardAxis},
+                                                               {"rotationAxis", m_rotationAxis},
+                                                               {"sideAxis", m_sideAxis},
+                                                               {"prepareButton", m_prepareButton},
+                                                               {"startButton", m_startButton},
+                                                               {"pauseButton", m_pauseButton},
+                                                               {"stopButton", m_stopButton},
+                                                               {"connectGoalButton", m_connectGoalButton},
+                                                               {"connectRpcButton", m_connectRpcButton},
+                                                               {"disconnectButton", m_disconnectButton}};
+    for (auto &input : joypadInputs)
```

Comment:
In principle, you could avoid parsing the buttons if it is a pedal

Reply:
done in [d08f09f](https://github.com/robotology/walking-controllers/pull/188/commits/d08f09fc2ea8c5a9580f82dc84760b4fcc39dc02)

Reply:
I think you deleted from the wrong config file 😉

Reply:
oh, reverting the wrong deletion! my bad